src/codeu/chat/client/commandline/Chat.java

```diff
@@ -457,4 +530,371 @@ public void invoke(List<String> args) {
     // so that it can be used.
     return panel;
   }
+  
+  private Panel createInterestPanel(final UserContext user) {
+
+    final Panel panel = new Panel();
+    
+    InterestSet interests = context.getInterestSet(user.user.id);
+    int updates = 0;
+    
+    ArrayList<Bookmark> display = new ArrayList<Bookmark>();
+    ArrayList<Time> unsorted = new ArrayList<Time>();
+    
+    for (User u : interests.users){
+      HashSet<ConversationContext> newSet = conversationsOfUser(u,user);
+      
+      for(ConversationContext c : newSet){
+        boolean isCopy = false;
+        for(Bookmark b : interests.bookmarks){
+          if(!isCopy){
+            if(b.conversation.title.equals(c.conversation.title)){
```

Comment:
It might be better to compare by `conversation.id` here, since they also uniquely identify conversations.